### PR TITLE
Gui controller

### PR DESCRIPTION
**Pull request by DierF**
Move update functionality to UML_Controller.
Add draw functions to be able to draw the diagram in each iteration of main loop.
Add related functions to GUI_View to simulate draw and listen.

### CODE REVIEW

File: src/Controllers/cli_controller.py [hunk 1]
```diff
@@ -4,25 +4,16 @@
...
 
-
-
 class CLI_Controller:
 
     def __init__(self):
         self._view = CLI_View()
     
-    
-    def update(self, parsed_input:list):
```
Comment (pwfreedm) [inline src/Controllers/cli_controller.py:15]: so... removing this bricks the CLI.

File: src/Controllers/controller.py [hunk 2]
```diff
@@ -26,7 +26,9 @@ def run(self):
         while not self._should_quit:
             try: 
                 data = self.parse(self._controller.request_update())
-                ret = self._controller.update(data)
+                ret = data[0](*data[1:])
+                if isinstance(ret, str):
+                    print(ret)
```
Comment (pwfreedm) [inline src/Controllers/controller.py:31]: this doesn't work because it will print in gui mode too.

Comment (pwfreedm) [reply src/Controllers/controller.py:31]: which we were explicitly warned against at the end of last sprint. This is why it was originally separated.

**Review by pwfreedm — CHANGES_REQUESTED** [review]

File: src/Controllers/cli_controller.py (hunk 1 above)
Comment (pwfreedm) [reply src/Controllers/cli_controller.py:15]: the idea was for both the gui and cli controllers to have an update method that equates to doing everything needed for a single command, then the main controller would just call update in a loop and everything else would happen regardless of which controller it possesses.

Comment (DierF) [reply src/Controllers/cli_controller.py:15]: Both controllers will do the same in update(), so I just put it back in UML_Controller. And I found it was redundant to make a separate method for the functionality, so I just put it back in run().

File: src/Controllers/controller.py (hunk 2 above)
Comment (DierF) [reply src/Controllers/controller.py:31]: Added a check for CLI mode

**Review by pwfreedm — APPROVED** [review]
kinda grumbly about this b/c unnecessary checks in the main loop happen more than anywhere else, but it's okay as long as it works.

File: src/Controllers/cli_controller.py (hunk 1 above)
Comment (DierF) [reply src/Controllers/cli_controller.py:15]: Yeah, this is a quick and probably temporary fix. For now, it's working. But we could add update() function back when we have different update functionality in CLI and GUI.

**Pull request merged**